src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.service.ReportGenerator;
+import java.util.stream.Collectors;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private static final String reportHead = "fruit,quantity";
+    private static final String reportDataDelimiter = ",";
+    private final FruitDao fruitDao;
+
+    public ReportGeneratorImpl(FruitDao fruitDao) {
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public String getReport() {
+        return reportHead + System.lineSeparator() + getReportData();
+    }
+
+    private String getReportData() {
```

Comment:
The method 'getReportData' is private and tightly coupled with 'getReport'. Consider if it's necessary to separate this logic into a private method, or if it could be integrated into 'getReport' for simplicity.

Reply:
++